rapids-cmake/cython/create_modules.cmake

```diff
@@ -0,0 +1,101 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+include_guard(GLOBAL)
+
+include(${CMAKE_CURRENT_LIST_DIR}/detail/verify_init.cmake)
+
+#[=======================================================================[.rst:
+rapids_cython_create_modules
+---------------------
+
+.. versionadded:: v22.06.00
+
+Generate C(++) from Cython and create Python modules.
+
+.. code-block:: cmake
+
+  rapids_cython_create_modules(<ModuleName...>)
+
+Creates a Cython target for a module, then adds a corresponding Python
+extension module.
+
+.. note::
+  Requires :cmake:command:`rapids_cython_init` to be called before usage.
+
+``CXX``
+  Flag indicating that the Cython files need to generate C++ rather than C.
+
+``SOURCE_FILES``
+  The list of Cython source files to be built into Python extension modules.
+  Note that this function assumes that Cython source files have a one-one
+  correspondence with extension modules to build, i.e. for every `<Name>.pyx`
+  in SOURCE_FILES we assume that `<Name>.pyx` is a Cython source file for which
+  an extension module `<Name>` should be built.
+
+``LINKED_LIBRARIES``
+  The list of libraries that need to be linked into all modules. In RAPIDS,
+  this list usually contains (at minimum) the corresponding C++ libraries.
+
+``INSTALL_ROOT``
+  The source directory of the project. This directory is used to compute the
```

Comment:
Am I correct in understanding this needs to be an absolute path? If so we should document that requirement here.

Reply:
Does it need to be? Does `cmake_path(RELATIVE_PATH <path-var> BASE_DIRECTORY base)` require that `base` be an absolute path, or can it also be relative (to the current directory)?

Reply:
No that supports relative paths.

I think the root issue is that I am confused on why this variable is needed. It is labelled INSTALL_ROOT, but we are asking for the source of the project. Do we mean the value that is held in [PROJECT_SOURCE_DIR](https://cmake.org/cmake/help/latest/variable/PROJECT_SOURCE_DIR.html)? 

Can you point me to an existing usage as that would be helpful

Reply:
I realized that you only get half the picture here. The relevant TODO item is on the corresponding raft PR [here](https://github.com/rapidsai/raft/pull/633/files#diff-e249c4fccd88ad8117d97abbcd719c9c2699ff75679c2e00d1def250132904bcR20). The variable is then used to determine the relative path to the install directory in this PR [here](https://github.com/rapidsai/rapids-cmake/pull/184/files#diff-528f59a9f5467b251f1dbafa7184b1bb1f93776a483b632da0bba7d7790d0b6dR96).